python/grass/jupyter/interactivemap.py

```diff
@@ -347,13 +353,161 @@ def add_layer_control(self, **kwargs):
+                print(f"North: {reprojected_region['north']}")
+                print(f"South: {reprojected_region['south']}")
+                print(f"East: {reprojected_region['east']}")
+                print(f"West: {reprojected_region['west']}")
+
+        def on_rectangle_change(value):
+            nonlocal region_coordinates, reprojected_region
+            latlon_bounds = value["new"][0]
+            region_coordinates = {
+                "north": latlon_bounds[2]["lat"],
+                "south": latlon_bounds[0]["lat"],
+                "east": latlon_bounds[2]["lng"],
+                "west": latlon_bounds[0]["lng"],
+            }
+            from_proj = "+proj=longlat +datum=WGS84 +no_defs"
+            to_proj = get_location_proj_string()
+            reprojected_region = reproject_region(
+                region_coordinates, from_proj, to_proj
+            )
+            update_region(
+                reprojected_region["north"],
+                reprojected_region["south"],
+                reprojected_region["east"],
+                reprojected_region["west"],
+            )
+
+        def toggle_region_mode(change):
+            nonlocal rectangle, save_button_control, region_coordinates
+
+            if change["new"]:
+                if rectangle is None:
+                    latlon_bounds = get_computational_region_bb()
+                    region_coordinates = {
+                        "North": latlon_bounds[1][0],
+                        "South": latlon_bounds[0][0],
+                        "East": latlon_bounds[1][1],
+                        "West": latlon_bounds[0][1],
+                    }
+                    rectangle = Rectangle(
+                        bounds=latlon_bounds,
+                        color="red",
+                        fill_color="red",
+                        fill_opacity=0.5,
+                        draggable=True,
+                        editable=True,
+                        transform=True,
+                        rotation=False,
+                        name="computational_region",
+                    )
+                    rectangle.observe(on_rectangle_change, names="locations")
+                    self.map.add_layer(rectangle)
+
+                else:
+                    latlon_bounds = rectangle.bounds
+                    rectangle.editable = True
+                    rectangle.draggable = True
+                    rectangle.transform = True
+                    rectangle.rotation = False
+
+                save_button.disabled = False
+                bottom_output_widget.layout.display = "block"
+
+                if save_button_control is None:
+                    save_button_control = WidgetControl(
+                        widget=save_button, position="topright"
+                    )
+                    self.map.add_control(save_button_control)
+                else:
+                    self.map.add_control(save_button_control)
+            else:
```

Comment:
The rectangle should disappear

Reply:
It's added to the layers, and if you just remove it from there, it'll disappear, but I would modify the code to make the rectangle disappear when the draw_computational_region button is toggled off.

Reply:
Hi Anna, I tried this in two ways:
1) Removing the rectangle using `self.map.remove(rectangle)` deletes the layer permanently.
2) Reduced the opacity to 0 so that the rectangle isn't visible, updated the code with this.

Reply:
I don't care that much which way it works, but when you toggle back to change the region, it should get the latest computational region. So for example:
1. Toggle region on
2. Change it and save.
3. Toggle region editing off (rectangle disappears)
4. In a different notebook cell, change region with g.region.
5. Toggle region on - right now the old rectangle shows up, it doesn't correspond to the new region, so this should be fixed.

Reply:
Hi Anna, 

For me, the new computational region shows up once I save the rectangle, toggle off the computational region button and turn it on again. If I rerun the cell, it shows me the modified region. Also, I have taken care of the region update in g.region; it updates the region in g.region when the user clicks save_region.

Reply:
Yes, but I am talking about a case when you don't rerun the cell.

Reply:
Yes, the region will also be updated at that time. I think there's a misunderstanding of what you are trying to convey and what I think you're saying.

Reply:
I got this. The rectangle bounds weren't updating. I have corrected this.

Reply:
I don't see how, it's still not working for me. You need to call `get_computational_region_bb` every time user toggles the region button on, so that the rectangle gets updated based on the currently set computational region.

Reply:
Here is the rewritten version:

In this line: [rectangle.bounds = latlon_bounds](https://github.com/OSGeo/grass/blob/a9d54909785b674187429602412c14d18be5bbf2/python/grass/jupyter/interactivemap.py#L418).

When I attempted to write code for zooming when the rectangle was already present [here](https://github.com/OSGeo/grass/blob/a9d54909785b674187429602412c14d18be5bbf2/python/grass/jupyter/interactivemap.py#L492-L497), I noticed that the coordinates weren't updating. However, after adding [rectangle.bounds = latlon_bounds](https://github.com/OSGeo/grass/blob/a9d54909785b674187429602412c14d18be5bbf2/python/grass/jupyter/interactivemap.py#L418), the rectangle updated every time, which I believe is the reason why it zooms in.